### PR TITLE
Add RISC-V architectures

### PR DESCRIPTION
**Pull request by diehlpk**
Add the two recently added RISC-V architectures to Kokkos.

### CODE REVIEW

**Comment by JBludau** [comment]
could you either hold this until https://github.com/kokkos/kokkos-core-wiki/pull/602 is merged and then add it in the corresponding table?

**Comment by diehlpk** [comment]
Yes, I can.

**Comment by JBludau** [comment]
@diehlpk #602 is merged now, please go ahead. Thanks for waiting

**Comment by diehlpk** [comment]
@JBludau I resolved the merge conflict but still need to wait for the merge conflict.

**Review by masterleinad — APPROVED** [review]

File: docs/source/keywords.rst [hunk 1]
```diff
@@ -487,14 +487,30 @@ If cross-compiling, or if you want to be specific, the CPU architecture can be p
     * - ``Kokkos_ARCH_KNL``
       - Knights Landing/x86-64
       - 31S1P @ Tianhe-2
-
```
Comment (dalg24) [inline docs/source/keywords.rst:490]: Please undo

File: docs/source/keywords.rst [hunk 2]
```diff
@@ -487,14 +487,30 @@ If cross-compiling, or if you want to be specific, the CPU architecture can be p
...
+    :header-rows: 1
+    :align: left
+    
+    * - CMake keyword
+      - Architecture/Instruction set
+      - Examples   
+    
+    * - ``KOKKOS_ARCH_RISCV_RVA22V``
+      -  RVA22V/RISC-V ISA
+      -
```
Comment (dalg24) [inline docs/source/keywords.rst:509]: No examples?

**Review by dalg24 — COMMENTED** [review]
Would you add a 4th column with "Notes" saying since what release these are available

File: docs/source/keywords.rst (hunk 2 above)
Comment (JBludau) [reply docs/source/keywords.rst:509]: and could you add a 4th column with the kokkos version that it was added in? see https://github.com/kokkos/kokkos-core-wiki/blob/main/docs/source/keywords.rst#architectures

Comment (diehlpk) [reply docs/source/keywords.rst:510]: I added the 4th column with the Kokkos version.

Comment (diehlpk) [reply docs/source/keywords.rst:510]: @dalg24 What should I put as examples? Just the hardware with this CPU?

File: docs/source/keywords.rst [hunk 3]
```diff
@@ -487,14 +487,33 @@ If cross-compiling, or if you want to be specific, the CPU architecture can be p
     * - ``Kokkos_ARCH_KNL``
       - Knights Landing/x86-64
       - 31S1P @ Tianhe-2
```
Comment (dalg24) [inline docs/source/keywords.rst]:
```suggestion
      - 31S1P @ Tianhe-2

```

File: docs/source/keywords.rst [hunk 4]
```diff
@@ -487,14 +487,33 @@ If cross-compiling, or if you want to be specific, the CPU architecture can be p
...
     * - ``Kokkos_ARCH_KNC``
       - Knights Corner/x86-64
       -
 
     * - ``Kokkos_ARCH_SNB``
       - Sandy Bridge/x86-64
       -
+      
+ .. list-table:: RISC-V CPU architectures
+    :widths: 30 30 30
```
Comment (dalg24) [inline docs/source/keywords.rst:500]:
```suggestion
    :widths: 30 30 30 30
```

File: docs/source/keywords.rst [hunk 5]
```diff
@@ -487,14 +487,33 @@ If cross-compiling, or if you want to be specific, the CPU architecture can be p
...
+      - Notes
+    
+    * - ``KOKKOS_ARCH_RISCV_RVA22V``
+      -  RVA22V/RISC-V ISA
+      -
+      - (since Kokkos 4.5.0)
+      
+    * - ``KOKKOS_ARCH_RISCV_SG2042``
+      -  SG2042/RISC-V ISA
+      -
```
Comment (dalg24) [inline docs/source/keywords.rst:516]:
```suggestion
      -  Milk-V's Pioneer
```

File: docs/source/keywords.rst [hunk 6]
```diff
@@ -487,14 +487,33 @@ If cross-compiling, or if you want to be specific, the CPU architecture can be p
...
+    
+    * - ``KOKKOS_ARCH_RISCV_RVA22V``
+      -  RVA22V/RISC-V ISA
+      -
+      - (since Kokkos 4.5.0)
+      
+    * - ``KOKKOS_ARCH_RISCV_SG2042``
+      -  SG2042/RISC-V ISA
+      -  
+      - (since Kokkos 4.3.0)
```
Comment (dalg24) [inline docs/source/keywords.rst:517]:
```suggestion
      - (since Kokkos 4.3)
```

File: docs/source/keywords.rst [hunk 7]
```diff
@@ -487,14 +487,33 @@ If cross-compiling, or if you want to be specific, the CPU architecture can be p
...
+    
+    * - CMake keyword
+      - Architecture/Instruction set
+      - Examples 
+      - Notes
+    
+    * - ``KOKKOS_ARCH_RISCV_RVA22V``
+      -  RVA22V/RISC-V ISA
+      -
+      - (since Kokkos 4.5.0)
```
Comment (dalg24) [inline docs/source/keywords.rst:512]:
```suggestion
      - (since Kokkos 4.5)
```

File: docs/source/keywords.rst (hunk 2 above)
Comment (dalg24) [reply docs/source/keywords.rst:511]:
```suggestion
      - SpacemiT K1
```

File: docs/source/keywords.rst [hunk 8]
```diff
@@ -495,6 +495,26 @@ If cross-compiling, or if you want to be specific, the CPU architecture can be p
...
+      - Notes
+    
+    * - ``KOKKOS_ARCH_RISCV_RVA22V``
+      -  RVA22V/RISC-V ISA
+      - SpacemiT K1
+      - (since Kokkos 4.5)
+      
+    * - ``KOKKOS_ARCH_RISCV_SG2042``
+      -  SG2042/RISC-V ISA
+      -  Milk-V Pioneer
```
Comment (dalg24) [inline docs/source/keywords.rst:516]:
```suggestion
      - SG2042/RISC-V ISA
      - Milk-V Pioneer
```

File: docs/source/keywords.rst [hunk 9]
```diff
@@ -495,6 +495,26 @@ If cross-compiling, or if you want to be specific, the CPU architecture can be p
...
+    :widths: 30 30 30 30
+    :header-rows: 1
+    :align: left
+    
+    * - CMake keyword
+      - Architecture/Instruction set
+      - Examples 
+      - Notes
+    
+    * - ``KOKKOS_ARCH_RISCV_RVA22V``
```
Comment (dalg24) [inline docs/source/keywords.rst:509]:
```suggestion
    * - ``Kokkos_ARCH_RISCV_RVA22V``
```

File: docs/source/keywords.rst [hunk 10]
```diff
@@ -495,6 +495,26 @@ If cross-compiling, or if you want to be specific, the CPU architecture can be p
...
+      - RVA22V/RISC-V ISA
+      - SpacemiT K1
+      - (since Kokkos 4.5)
+      
+    * - ``KOKKOS_ARCH_RISCV_SG2042``
```
Comment (dalg24) [inline docs/source/keywords.rst:514]:
```suggestion
    * - ``Kokkos_ARCH_RISCV_SG2042``
```

**Comment by dalg24** [comment]
Closing in favor of #608
Wasted enough time trying to fiddle with it on the online editor

**Pull request closed without merging**